src/v1/sources/shopify/webhookTransformations/serverSideTransform.js

```diff
@@ -125,6 +127,13 @@ const processEvent = async (inputEvent, metricMetadata) => {
     const anonymousId = getAnonymousIdFromAttributes(event);
     if (isDefinedAndNotNull(anonymousId)) {
       message.setProperty('anonymousId', anonymousId);
+    } else {
+      // if anonymousId is not present in note_attributes or note_attributes is not present, query redis for anonymousId
+      const cartToken = getCartToken(event, message);
```

Comment:
redis call should be made only if valid cartToken exists
```suggestion
      const cartToken = event?.cart_token;
```
do we need getCartToken function ?

Reply:
Updated.
Keeping it in a separate function gives better readability to the functionality and also if any additional related logic comes up, can be abstracted to the function.